### PR TITLE
OPENEUROPA-2391: Switch between the EC and EU footer blocks.

### PR DESCRIPTION
**Pull request by ademarco**

### CODE REVIEW

File: modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    $component_library = $this->configFactory->get('oe_theme.settings')->get('component_library');
+    return $component_library === $this->configuration['component_library'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    if (empty($this->configuration['component_library'])) {
+      return $this->t('The current component library is set to any value.');
```
Comment (nagyad) [inline modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php]: I think it is better if we say "The current component library is not set."

File: modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      '#title' => $this->t('Component library'),
+      '#options' => [
+        '' => $this->t('- Any -'),
+        'ec' => $this->t('European Commission'),
+        'eu' => $this->t('European Union'),
+      ],
+      '#default_value' => $this->configuration['component_library'],
+      '#description' => t('Choose with which component library this condition should be met.'),
+    ];
+    return parent::buildConfigurationForm($form, $form_state);
```
Comment (nagyad) [inline modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php:90]: Empty line before return statement.

File: tests/Behat/OeThemeTestContext.php [hunk 3]
```diff
@@ -276,4 +276,88 @@ protected function getDemoPageByTitle(string $title): NodeInterface {
...
+    // time.
+    //
+    // We have a step in EWCMS that does the same thing, ideally we would need
+    // to port this in our traits and remove it from here.
+    // @todo: reuse reset check sums once available as a trait.
+    \Drupal::service('cache_tags.invalidator')->resetCheckSums();
+  }
+
+  /**
+   * Assert presence of absence of footer blocks.
```
Comment (nagyad) [inline tests/Behat/OeThemeTestContext.php]: Assert presence or absence of footer blocks.

File: tests/features/corporate-blocks.feature [hunk 4]
```diff
@@ -4,24 +4,43 @@ Feature: Corporate blocks feature
...
-    And I should see "European Union" in the "footer" element
+    Then I should see "European Commission" footer
 
     Examples:
       | page                        |
-      | the homepage                |
+      | the home page               |
+      | the user registration page  |
+
+  Scenario Outline: The European Commission footer or the European Union one si shown depending on the which style is chosen.
```
Comment (nagyad) [inline tests/features/corporate-blocks.feature]: Typo in sentence.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Comment by nagyad** [comment]
Tested manually both upgrade path and the functionality.

File: oe_theme.info.yml [hunk 5]
```diff
@@ -31,6 +31,7 @@ config_devel:
     - responsive_image.styles.oe_theme_main_content
   optional:
     - block.block.oe_theme_corporate_footer
```
Comment (sergepavle) [inline oe_theme.info.yml]: Should be removed/replaced to     `- block.block.oe_theme_ec_corporate_footer`.

**Review by sergepavle — CHANGES_REQUESTED** [review]

File: modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a configurable 'Current component library' condition.
+ *
+ * This condition check if the current component library is equal to a given
```
Comment (imanoleguskiza) [inline modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php]: "This condition checks"

File: tests/Behat/OeThemeTestContext.php [hunk 7]
```diff
@@ -276,4 +276,88 @@ protected function getDemoPageByTitle(string $title): NodeInterface {
     return reset($nodes);
   }
 
+  /**
+   * Assert given corporate footer presence on page.
+   *
+   * @Then I should see :component_library footer( instead)
```
Comment (imanoleguskiza) [inline tests/Behat/OeThemeTestContext.php]: MIssing the article "I should see the :component_library"

File: tests/Behat/OeThemeTestContext.php [hunk 8]
```diff
@@ -276,4 +276,88 @@ protected function getDemoPageByTitle(string $title): NodeInterface {
...
+   */
+  public function assertFooterBlockOnPage(string $component_library): void {
+    $this->assertFooter($component_library, TRUE);
+  }
+
+  /**
+   * Assert given corporate footer absence on page.
+   *
+   * @Then I should not see :component_library footer( instead)
```
Comment (imanoleguskiza) [inline tests/Behat/OeThemeTestContext.php]: MIssing the article "I should see the :component_library"

File: tests/Behat/OeThemeTestContext.php [hunk 9]
```diff
@@ -276,4 +276,88 @@ protected function getDemoPageByTitle(string $title): NodeInterface {
...
+    $this->assertFooter($component_library, FALSE);
+  }
+
+  /**
+   * Set theme's corporate library value.
+   *
+   * @Given the theme is configured to use the :component_library style
+   */
+  public function setCorporateLibrary(string $component_library): void {
+    $component_library_name = [
```
Comment (imanoleguskiza) [inline tests/Behat/OeThemeTestContext.php:303]: Should we define this as a constant or a class variable and reuse it in all the methods?

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

File: tests/Behat/OeThemeTestContext.php (hunk 9 above)
Comment (ademarco) [reply tests/Behat/OeThemeTestContext.php:303]: I wouldn't, it's specific to converting human readable configuration into its machine value.

File: modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php (hunk 1 above)
Comment (ademarco) [reply modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php]: That means that the condition is satisfied for any value of the component library.

Comment (ademarco) [reply modules/oe_theme_helper/src/Plugin/Condition/CurrentComponentLibraryCondition.php]: I've changed that to a more clear:
> The current component library can be set to anything

**Review by sergepavle — DISMISSED** [review]

**Review by sergepavle — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Pull request merged**